integrations/cri_apis/cri_apis.go

```diff
@@ -1,65 +1,112 @@
 package cri_apis
 
 import (
+	"context"
 	"fmt"
-	"log"
+	"os"
+	"time"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials/insecure"
+	"github.com/postmanlabs/postman-insights-agent/printer"
 	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
 )
 
-const containerdCRIEndpoint = "unix:///var/run/containerd/containerd.sock"
-const CRIOEndpoint = "unix:///var/run/crio/crio.sock"
+const (
+	// Context timeout for all CRI operations
+	connectionTimeout = 5 * time.Second
 
-// grpc library default is 4MB for message size
-const maxMsgSize = 1024 * 1024 * 4
-
-type NetworkNamespace string
-
-type ContainerInfo struct {
-	// Container Info struct
-}
+	// Default containerd runtime endpoint
+	containerdCRIEndpoint = "unix:///run/containerd/containerd.sock"
+)
 
+// CriClient struct holds the runtime service client
 type CriClient struct {
-	// CRI Client struct
+	runtimeService *remoteRuntimeService
 }
 
-// Constructor to init the client
+// NewCRIClient initializes a new CRI client
 func NewCRIClient() (*CriClient, error) {
-	// Set up connection to CRI
-	conn, err := grpc.Dial(
-		containerdCRIEndpoint, // CRI Endpoint
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxMsgSize)),
+	var (
+		service *remoteRuntimeService
+		err     error
 	)
-	if err != nil {
-		log.Fatalf("Failed to connect to CRI endpoint: %v", err)
+
+	criEndpoint := os.Getenv("POSTMAN_CRI_ENDPOINT")
+	if criEndpoint != "" {
+		service, err = newRemoteRuntimeService(criEndpoint, connectionTimeout)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		printer.Infoln("No CRI endpoint provided, trying default CRI endpoint")
+		service, err = newRemoteRuntimeService(containerdCRIEndpoint, connectionTimeout)
+		if err != nil {
+			printer.Errorf("Failed to connect to %s: %v\n", containerdCRIEndpoint, err)
+		}
+	}
+
+	if service == nil {
+		return nil, fmt.Errorf("failed to connect to CRI endpoint")
 	}
-	defer conn.Close()
 
-	// Create runtime service client
-	// TODO: assign runtime to CriClient struct
-	_ = runtime.NewRuntimeServiceClient(conn)
+	criClient := CriClient{
+		runtimeService: service,
+	}
 
-	return &CriClient{}, nil
+	return &criClient, nil
 }
 
-// Function to inspect the container with the given ID. This will be the internal function ??
+// inspectContainer inspects the container with the given ID
 func (cc *CriClient) inspectContainer(containerID string) (ContainerInfo, error) {
+	containerStatusRequest := &runtime.ContainerStatusRequest{
+		ContainerId: containerID,
+		Verbose:     true, // Needed to get info object
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), cc.runtimeService.timeout)
+	defer cancel()
```

Comment:
Probably should have flagged this earlier -- the right thing to do is probably get a context from the caller (which you can then fork off a new one).  That way if the outer context is cancelled this one will be too.